### ISSUE TITLE
Wildcard listener not invoked when also matching exact listener

### ISSUE THREAD

**Issue by thorsby** [EventEmitter2/EventEmitter2#278]
If an event matches both a listener with a wildcard and a listener without, only the non-wildcard listener is invoked.
A wildcard listener should not be dependent on if another listener has a "better match". If this is actually intended behaviour, maybe an option on the constructor could provide both.

**Steps to reproduce:**
```
const EventEmitter2 = require('eventemitter2');
const ee = new EventEmitter2({ wildcard: true });

ee.on('A', console.log);
ee.on('B.**', console.log);
ee.on('C', console.log);
ee.on('C.**', console.log);

ee.emit('A', 'A'); // Logs "A" once
ee.emit('B', 'B'); // Logs "B" once
ee.emit('C', 'C'); // Logs "C" only once, not matching wildcard C.**
```

**Expected result**
A
B
C
C

Actual result
A
B
C

**Comment by Smiche** [comment]
`ee.emit('C', 'C'); // Logs "C" only once, not matching wildcard C.**`
That seems like correct behavior to me? Dot is a delimeter, `C.**` will match everything after `C.`

**Comment by thorsby** [comment]
`C.**` should match the emitted **C**,  just like the above `B.**` matches **B**
`C.**` also does match, if the exact `C` is removed, but when both listeners are added, only one (the exact topic and not the wildcard) gets invoked.

**Comment by Smiche** [comment]
Yup, just noticed that. That is very inconsistent behavior.

**Comment by DigitalBrainJS** [comment]
@thorsby Thanks for reporting! I have just fixed this bug and when @RangerMauve has time he will release a new v6.4.5 version with this change.

**Comment by RangerMauve** [comment]
@DigitalBrainJS Thank you! Just released v6.4.5 on NPM. 😁
